packages/features/ee/payments/components/Payment.tsx

```diff
@@ -120,15 +119,21 @@ export default function PaymentComponent(props: Props) {
       });
     }
   };
+
+  const handleChange = async (event: StripeCardElementChangeEvent) => {
+    // Listen for changes in the CardElement
+    // and display any errors as the customer types their card details
+    setState({ status: "idle" });
+    if (event.error)
+      setState({ status: "error", error: new Error(event.error?.message || t("missing_card_fields")) });
+  };
+
   return (
     <form id="payment-form" className="bg-subtle mt-4 rounded-md p-6" onSubmit={handleSubmit}>
-      <p className="font-semibold">{t("card_details")}</p>
-      <CardElement
-        className="my-5 bg-white p-2"
-        id="card-element"
-        options={CARD_OPTIONS}
-        onChange={handleChange}
-      />
+      <div>
+        <PaymentElement onChange={() => setState({ status: "idle" })} />
```

Comment:
Stripe recommends using `PaymentElement`. The `onChange` is to clear any errors and enable the submit button once changes are made after an error